### ISSUE TITLE
Handling version numbers and ref sorting

### ISSUE THREAD

**Issue by djhoese** [Holzhaus/sphinx-multiversion#50]
Currently the references are sorted based on remote (it is a remote or local reference), but this doesn't take into consideration the date or refname of the references. Where this is a problem is with tags that are semantic versions. For example, in my project I have tags `v0.7.0` and `v0.7.1`. By the default sorting of `sorted` this will process `v0.7.0` first. If I identify this version as `0.7` then that means the `0.7` version of my documentation will be based on the older tag which I assume is undesirable in most cases. I think a simple fix might be (reverse the sort and negate the is_remote):

```
    if config.smv_prefer_remote_refs:
        gitrefs = sorted(gitrefs, key=lambda x: (x.is_remote, x.refname, *x), reverse=True)
    else:
        gitrefs = sorted(gitrefs, key=lambda x: (not x.is_remote, x.refname, *x), reverse=True)
```

**Comment by tfoote** [comment]
We have a use case where we're using codenames and not specific versions. There's currently a debate about whether to sort it forwards or reverse alphabetically. It would be great if there was a way to control or override the sort generally depending on the preference of the developer/documentation writer. Just changing the default will make it work better for some cases and worse for others.

**Comment by djhoese** [comment]
Yeah understood. In the case of code names though, each code name is resolved to a separate output version/directory, right? In the case above my 0.7.x versions need *only* the newest one to be created. Currently this doesn't work.

Of course having options for all of this would be nice.
